File: backend/src/main/java/com/hungrylab/backend/Controller/AuthController.java
package com.hungrylab.backend.Controller;

import com.hungrylab.backend.Service.AuthService;
import com.hungrylab.backend.dto.AuthResponseDto;
import com.hungrylab.backend.dto.LoginRequestDto;
import com.hungrylab.backend.dto.RegisterRequestDto;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/auth")
public class AuthController {

    private final AuthService authService;

    public AuthController(AuthService authService) {
        this.authService = authService;
    }

    @PostMapping("/register")
    public ResponseEntity<?> register(@RequestBody RegisterRequestDto request) {
        try {
            return ResponseEntity.ok(authService.register(request));
        } catch (Exception e) {
            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(e.getMessage());
        }
    }

    @PostMapping("/login")
    public ResponseEntity<?> login(@RequestBody LoginRequestDto request) {
        try {
            return ResponseEntity.ok(authService.authenticate(request));
        } catch (Exception e) {
            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body("Invalid credentials");
        }
    }

    // TEMPORARY UTILITY: Hit this in your browser to upgrade an account without pgAdmin!
    @GetMapping("/make-admin")
    public ResponseEntity<String> makeAdmin(@RequestParam String email) {
        return ResponseEntity.ok(authService.upgradeToAdmin(email));
    }
}

File: express-backend/src/controllers/authController.ts
import { Request, Response } from 'express';
import * as authService from '../services/authService';

export const register = async (req: Request, res: Response) => {
    try {
        const result = await authService.register(req.body);
        res.json(result);
    } catch (e: any) {
        res.status(400).json(e.message);
    }
};

export const login = async (req: Request, res: Response) => {
    try {
        const result = await authService.authenticate(req.body);
        res.json(result);
    } catch (e) {
        res.status(401).json("Invalid credentials");
    }
};

export const makeAdmin = async (req: Request, res: Response) => {
    try {
        const { email } = req.query;
        const msg = await authService.upgradeToAdmin(email as string);
        res.json(msg);
    } catch (e: any) {
        res.status(400).json(e.message);
    }
};
